rct229/ruleset_functions/compare_schedules.py

```diff
@@ -0,0 +1,68 @@
+from rct229.utils.assertions import RCTFailureException
+
+REGULAR_YEAR_HOUR = 8760
+LEAP_YEAR_HOUR = 8784
+
+
+def compare_schedules(schedule_1: list, schedule_2: list, mask_schedule: list, comparison_factor: float, is_leap_year: bool):
```

Comment:
It might be convenient to default 
- comparison_factor (default to 1)
- is_leap_year (default to False)
- mask_schedule (all ones; would have to set this in the function body if mask_schedule is None; for that reason is should be the last argument). Defaulting this one is somewhat questionable to me, but the other two seems like very good candidates.

Reply:
> It might be convenient to default
> 
> * comparison_factor (default to 1)
> * is_leap_year (default to False)
> * mask_schedule (all ones; would have to set this in the function body if mask_schedule is None; for that reason is should be the last argument). Defaulting this one is somewhat questionable to me, but the other two seems like very good candidates.

comparison_factor - agree, 1
is_leap_year - this shall come from RDS, I was thinking that this shall be taken from RDS, so I incline not to add default value, but it is probably good to discuss.
mask_schedule - probably a good idea to give default

I reviewed a few RDSs that have dependency on this function, it seems all mask_schedule is 1s' and comparison factor set to 1...